### ISSUE TITLE
lakeFS hadoop client fails to delete Spark staging directory

### ISSUE THREAD

**Issue by itaiad200** [treeverse/lakeFS#2797]
For lakeFS hadoop client version 0.1.4, the empty directory prefix (the object that ends with a `/`) that represents the staging directory, isn't deleted on successful completion of the job. This does not reproduce on the same spark+hadoop versions but with client version 0.1.1. Logging in the lakeFS server side does not suggest failure to delete that object.

**Comment by talSofer** [comment]
@itaiad200 can you please add the steps to reproduce? 
I'm guessing that they are: 
1. configure spark to work with lakefsfs 0.1.4 
2. write a file into lakefs
3. check the lakefs repo and see and empty "/" directory?

**Comment by itaiad200** [comment]
@talSofer - Right. I would suggest writing more than a single "file". It should be written using the OutputCommitter so that a staging directory will be created.

**Comment by talSofer** [comment]
I'm unable to reproduce this. here is what I tried: 
* writing a partitioned parquet into lakeFS with spark 2.4.7 and HadoopFS 0.1.4
* writing a partitioned parquet into lakeFS with spark 3.1.1 and HadoopFS 0.1.4

Checked the lakeFS repository after the job was completed, and it looks as expected.
![Screen Shot 2022-02-21 at 9 58 23](https://user-images.githubusercontent.com/14786381/154912328-49f881d8-7b2f-4c13-b1de-8ae7f794f106.png)

@itaiad200 any pointers to something else that you did when you saw the issue?
